sycl/source/device.cpp

```diff
@@ -60,6 +60,51 @@ vector_class<device> device::get_devices(info::device_type deviceType) {
       }
     }
   }
+
+  // If SYCL_BE is set and there are multiple devices of the same type
```

Comment:
I'm confused with meaning of SYCL_BE var. In another PR you are making so that SYCL_BE makes some preference towards specified BE in the selectors, but this patch removes all devices that do not match specified BE.
The documentation says that this var controls which BE will be used when a device is chosen using default selector only.
I think we should have consistent behavior and update documentation. Probably we need to introduce new env var if we need two different behaviors(force and preference).